core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -117,4 +124,24 @@ void handleFavicon(HttpRequest request, HttpChannel channel) {
             channel.sendResponse(new BytesRestResponse(FORBIDDEN));
         }
     }
+
+    private static final class ResourceHandlingHttpChannel extends HttpChannel {
+        private final HttpChannel delegate;
+        private final HttpServerTransport transport;
+
+        public ResourceHandlingHttpChannel(HttpChannel delegate, HttpServerTransport transport) {
+            super(delegate.request(), delegate.detailedErrorsEnabled());
```

Comment:
You're right! When we think about API evolution I think it is dangerous that `ResourceHandlingHttpChannel` extends from a class (`HttpChannel`) instead of implementing an interface. I think introducing an interface for `RestChannel` (or turning `RestChannel` itself into an interface) is a good longer term solution as we need to enforce that we delegate all methods and when the interface changes we get compile errors. If we stick with `RestChannel` as a base class we can easily add a new method to the base class and miss a delegation. Wdyt?